test/infrastructure/docker/e2e/docker_test.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/bootstrap/kubeadm/types/v1beta1"
+	"sigs.k8s.io/cluster-api/test/framework"
+
+	corev1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha2"
+	infrav1 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1alpha2"
+)
+
+var _ = Describe("Docker", func() {
+	Describe("Cluster Creation", func() {
+
+		namespace := "default"
+
+		clusterGen := &ClusterGenerator{}
+		nodeGen := &NodeGenerator{}
+
+		FContext("One node cluster", func() {
+			It("should create a single node cluster", func() {
+				cluster, infraCluster := clusterGen.GenerateCluster(namespace)
+				singleNode := nodeGen.GenerateNode(cluster.GetName())
+				input := &framework.OneNodeClusterInput{
+					Management:    mgmt,
+					Cluster:       cluster,
+					InfraCluster:  infraCluster,
+					Node:          singleNode,
+					CreateTimeout: 3 * time.Minute,
+				}
+
+				framework.OneNodeCluster(input)
+
+				cleanupInput := &framework.CleanUpInput{
+					Management: mgmt,
+					Cluster:    cluster,
+				}
+
+				framework.CleanUp(cleanupInput)
+			})
+		})
+
+		FContext("Multi-node controlplane cluster", func() {
```

Comment:
do you need the `Focus` here on both Contexts?

Reply:
Yep, the focus allows me to run a specific set of tests based on a regex

Reply:
so eventually when it's ready to merge wouldn't you remove that? you can the regex from the command line without the need for the `F` here I believe

Reply:
Ah, got it, went through the docs and I see what you're saying, I'll pull them out 👍 thanks